packages/language/src/workspace/plugin-configuration-provider.ts

```diff
@@ -121,6 +128,8 @@ export function deserializeProcessGroup(
   const libs = obj.libs || [];
   const lspOptions = obj["lsp-options"] || {};
   const checkMargins = lspOptions["check-margins"] ?? false;
+  const instructionCounterLimit =
+    lspOptions["instruction-counter-limit"] ?? 5000;
```

Comment:
Suggestion: Since we have an `isNumber` check, we don't need the null check here.
```suggestion
  const instructionCounterLimit =
    lspOptions["instruction-counter-limit"];
```